### PR TITLE
Enable building ARM64 version of this library

### PR DESCRIPTION
**Pull request by Alan-Jowett**
This pull request includes significant updates to the build configurations and project files to support multiple architectures (x64 and ARM64). The changes involve modifying the build matrix in GitHub Actions workflows, updating project configuration files, and ensuring conditional compilation settings for different platforms.

### GitHub Actions Workflow Updates:
* [`.github/workflows/build.yml`](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R23-R31): Added `architecture` to the build matrix and updated environment variables and commands to support both `x64` and `ARM64` architectures. Conditional steps were added to ensure certain tests only run on `x64`. [[1]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R23-R31) [[2]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L49-R50) [[3]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R59-R80) [[4]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R92) [[5]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L115-R119) [[6]](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R129-R150)

### Project Configuration Updates:
* [`Directory.Build.props`](diffhunk://#diff-9da24614831c308827a1ae533ffea392c97638c261dd42bd0f5226baa136d16eR18-R66): Added project configurations for `Debug`, `Release`, and `FuzzerDebug` for `Win32`, `x64`, and `ARM64` platforms. Also included conditional compilation settings based on the platform.

### Project File Updates:
* [`cxplat/cxplat_test/cxplat_test.vcxproj`](diffhunk://#diff-35ca574a84a5c3a69a8c247a377f4a9a59fa89eb68b65b69a349ef12236fb395L3-L20): Removed platform-specific configurations and replaced them with general configurations for `Debug`, `Release`, and `FuzzerDebug`. Simplified the property groups and item definition groups to be platform-agnostic. [[1]](diffhunk://#diff-35ca574a84a5c3a69a8c247a377f4a9a59fa89eb68b65b69a349ef12236fb395L3-L20) [[2]](diffhunk://#diff-35ca574a84a5c3a69a8c247a377f4a9a59fa89eb68b65b69a349ef12236fb395L29-R17) [[3]](diffhunk://#diff-35ca574a84a5c3a69a8c247a377f4a9a59fa89eb68b65b69a349ef12236fb395L60-R36) [[4]](diffhunk://#diff-35ca574a84a5c3a69a8c247a377f4a9a59fa89eb68b65b69a349ef12236fb395L114-R49)
* [`cxplat/src/cxplat_winkernel/cxplat_winkernel.vcxproj`](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL7-L32): Similar to `cxplat_test.vcxproj`, removed platform-specific configurations and replaced them with general configurations. Updated item definition groups to support different platforms conditionally. [[1]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL7-L32) [[2]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL67-R41) [[3]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL77-R51) [[4]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL87-R61) [[5]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL96-R77) [[6]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL142-R89) [[7]](diffhunk://#diff-694ef7fbe23ed43a03aa00027632467b354d6914bad57aede76364625c939e4bL154-R101)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,15 +20,17 @@ jobs:
...
     env:
       # Configuration type to build.
       # You can convert this to a build matrix if you need coverage of multiple configuration types.
       # https://docs.github.com/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
       BUILD_CONFIGURATION: ${{matrix.configurations}}
 
-      BUILD_PLATFORM: x64
+      BUILD_PLATFORM: ${{matrix.architecture}}
       CXPLAT_MEMORY_LEAK_DETECTION: true
+      PLATFORM_TOOLSET: v143
```
Comment (nibanks) [inline .github/workflows/build.yml:33]: Why do you need to specify the toolset explicitly?

File: sample/sample.vcxproj [hunk 2]
```diff
@@ -1,51 +1,32 @@
...
+  <PropertyGroup Condition="'$(Configuration)'=='Release'" Label="Configuration">
     <TargetVersion>Windows10</TargetVersion>
     <UseDebugLibraries>false</UseDebugLibraries>
     <PlatformToolset>v143</PlatformToolset>
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <DriverType>KMDF</DriverType>
     <DriverTargetPlatform>Universal</DriverTargetPlatform>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'" Label="Configuration">
+  <!-- <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|ARM64'" Label="Configuration">
```
Comment (nibanks) [inline sample/sample.vcxproj:29]: Why comment out and not just delete?

**Review by nibanks — APPROVED** [review]
Need to update the branch policy now because of the job name changes.

File: sample/sample.vcxproj (hunk 2 above)
Comment (Alan-Jowett) [reply sample/sample.vcxproj]: Mistake on my part. I was commenting it out while working on this, forgot to delete it.

**Pull request merged**